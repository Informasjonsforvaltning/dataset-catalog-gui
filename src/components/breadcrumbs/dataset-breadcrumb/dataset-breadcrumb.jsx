import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { getDatasetItemByDatasetiId } from '../../../redux/modules/datasets';
import { getTranslateText } from '../../../services/translateText';
import localization from '../../../services/localization';

export const PuredatasetBreadcrumb = props => {
  const { datasetItem } = props;
  return (
    <span>
      {getTranslateText(_.get(datasetItem, 'title')) ||
        localization.breadcrumbs.datasetRegistration}
    </span>
  );
};

const mapStateToProps = ({ datasets }, ownProps) => {
  const catalogId = _.get(ownProps, ['match', 'params', 'catalogId']);
  const id = _.get(ownProps, ['match', 'params', 'id']);

  const datasetItem = getDatasetItemByDatasetiId(datasets, catalogId, id) || {
    datasetItem: null
  };

  return {
    datasetItem
  };
};

PuredatasetBreadcrumb.defaultProps = {
  datasetItem: null
};

PuredatasetBreadcrumb.propTypes = {
  datasetItem: PropTypes.object
};

export const DatasetBreadcrumb = connect(mapStateToProps)(
  PuredatasetBreadcrumb
);
