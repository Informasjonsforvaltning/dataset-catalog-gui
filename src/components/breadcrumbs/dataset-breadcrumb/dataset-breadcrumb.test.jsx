import React from 'react';
import { shallow } from 'enzyme';
import { PuredatasetBreadcrumb } from './dataset-breadcrumb';

test('should render PuredatasetBreadcrumb with "dataset" active correctly', () => {
  const defaultProps = {
    datasetItem: {
      title: 'dataset title'
    }
  };
  const wrapper = shallow(<PuredatasetBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PuredatasetBreadcrumb correctly with no datasetItem', () => {
  const defaultProps = {};
  const wrapper = shallow(<PuredatasetBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
