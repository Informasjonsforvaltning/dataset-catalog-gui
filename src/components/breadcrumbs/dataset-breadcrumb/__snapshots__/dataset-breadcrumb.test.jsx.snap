// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `should render PuredatasetBreadcrumb correctly with no datasetItem 1`
] = `
<span>
  Registrer ny datasettbeskrivelse
</span>
`;

exports[
  `should render PuredatasetBreadcrumb with "dataset" active correctly 1`
] = `
<span>
  dataset title
</span>
`;
